### DISCUSSION TITLE
How to search array in array with dynamic variables using yq

### DISCUSSION THREAD

**Discussion by dnyaneshwarjadhav** [mikefarah/yq#1255] [Q&A]
I am having json file as
**sample.json**
```
[
  {
    "id": "1",
    "name": "name1",
  },
  {
    "id": "2",
    "name": "name2",
  },
  {
    "id": "3",
    "name": "name3",
  }
]
```

Another YAML file(target.yaml)
```
-
  name: default
  webhookurl: "http://localost:9091/te"
-
  name: default1
  webhookurl: "http://localost:9091/tee"
```

I wanted to get final result as data from YAML file and perform contains operation on data from **sample.json** file.
I tried with following way
 `yq e '. | .[].name | contains("default","default1")'` **sample.json**
This is giving result as follow:-
_$ yq e '. | .[].name | contains("default","default1")' myres.txt
false
false
false
false
true
false_

Another approach tries is
`myarr=$(yq e '. | .[].name ' target.yaml)` # Put array into **myarr** variable
`yq e '. | .[].name | contains(myarr)' sample.json` # This wont work
`yq e '. | .[].name | contains(env(myarr))' sample.json`  # This wont work either

**Expecting result should be**
```
default -> true
default1 -> false
```

**Comment by dnyaneshwarjadhav** [comment]
@mikefarah  any suggestions on this ?

**Comment by mikefarah** [comment]
I'm not 100% sure what you are trying to do - I think you are trying to find entries in target.yml that have a matching name from sample.yml?

I updated your sample.yaml to include matching data:

sample.yaml
```
[
  {
    "id": "1",
    "name": "default1",
  },{
    "name": "default2",
  }
]
```

so that:


```bash
yq '.[] | select( .name == (load("sample.yaml") | .[].name )  )' target.yaml
```

results in:
```
name: default1
webhookurl: "http://localost:9091/tee"
```

Explanation:
- We load in the target.yml (the file we want to filter)
- `(load("sample.yaml") | .[].name )` will load sample.yaml for the filter data, and process it into a set of matching values we want.
- select all the entries where the name matches one of those matching values
